scripts/src/blockchain/transactionBuilder.ts

```diff
@@ -25,8 +34,8 @@ export class TransactionBuilder {
 		return this;
 	}
 
-	public addMemo(memo: string): this {
-		(this as any).memo = memo;
+	public addMemo(memo: Memo<MemoType.Text> | undefined, appId: string): this {
+		(this as any)._transactionBuilder.memo = memo ? Memo.text('1-' + appId + '-' + memo.value) : Memo.none();
```

Comment:
is the string concatenation safe in the case of an undefined memo?

Reply:
There is a check if the memo is undefined.